common/autoware_test_utils/include/autoware_test_utils/autoware_test_utils.hpp

```diff
@@ -512,6 +513,62 @@ void publishToTargetNode(
   autoware::test_utils::spinSomeNodes(test_node, target_node, repeat_count);
 }
 
+/**
+ * @brief Manages publishing and subscribing to ROS topics for testing Autoware.
+ *
+ * The AutowareTestManager class provides utility functions to facilitate
+ * the publishing of messages to specified topics and the setting up of
+ * subscribers to listen for messages on specified topics. This class
+ * simplifies the setup of test environments in Autoware.
+ */
+class AutowareTestManager
+{
+public:
+  AutowareTestManager()
+  {
+    test_node_ = std::make_shared<rclcpp::Node>("autoware_test_manager_node");
+  }
+
+  template <typename MessageType>
+  void test_pub_msg(
+    rclcpp::Node::SharedPtr target_node, const std::string & topic_name, MessageType & msg)
+  {
+    if (publishers_.find(topic_name) == publishers_.end()) {
+      auto publisher = test_node_->create_publisher<MessageType>(topic_name, 10);
+      publishers_[topic_name] = std::static_pointer_cast<void>(publisher);
+    }
+
+    auto publisher =
```

Comment:
2. And use `std::dynamic_pointer_cast` to downcast from PublisherBase

Reply:
Fixed: https://github.com/autowarefoundation/autoware.universe/pull/7597/commits/4e9c1cadd093f6cdc1333b1d1e707271c08c5b16